### PR TITLE
DA-2501 : Add GetIdentity API 

### PR DESCRIPTION
**Pull request by linonymous**
Add the api to get the identity associated with the uuid.

### CODE REVIEW

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm

**Comment by lukaszgryglicki** [comment]
@linonymous can you please also add an example script that uses this API, something like [this](https://github.com/LF-Engineering/dev-analytics-affiliation/blob/master/sh/curl_get_profile.sh) and also update the main [README.md](https://github.com/LF-Engineering/dev-analytics-affiliation/blob/master/README.md#start-api-server-using) (just add an example how to call this API).

**Comment by lukaszgryglicki** [comment]
Hey, I've just noticed that you want identity by `uuid` not `id`.
Note that `GetIdentity` returns identity with a given `id` not `uuid`: https://github.com/LF-Engineering/dev-analytics-affiliation/blob/efcfd70101c462d9ae26f1e44c3791add13a9877/shdb/service.go#L1741
To get identity using UUID you will have to write another function in `shdb/service.go` and also note that there can be multiple identities with the same uuid.

**Comment by linonymous** [comment]
> @linonymous can you please also add an example script that uses this API, something like [this](https://github.com/LF-Engineering/dev-analytics-affiliation/blob/master/sh/curl_get_profile.sh) and also update the main [README.md](https://github.com/LF-Engineering/dev-analytics-affiliation/blob/master/README.md#start-api-server-using) (just add an example how to call this API).

Done! Added the script and README.md

**Comment by linonymous** [comment]
> Hey, I've just noticed that you want identity by `uuid` not `id`.
> Note that `GetIdentity` returns identity with a given `id` not `uuid`:
> 
> https://github.com/LF-Engineering/dev-analytics-affiliation/blob/efcfd70101c462d9ae26f1e44c3791add13a9877/shdb/service.go#L1741
> 
> 
> To get identity using UUID you will have to write another function in `shdb/service.go` and also note that there can be multiple identities with the same uuid.

Yes. The objective is to query the id only. So we can leverage the existing function. To get rid of the confusion, I have updated the separate input as ID instead of UUID.

File: affiliation/service.go [hunk 1]
```diff
@@ -1983,6 +1987,27 @@ func (s *service) GetListProfiles(ctx context.Context, params *affiliation.GetLi
 	return
 }
 
+// GetIdentity: API params:
+// /v1/affiliation/get_identity/{uuid}
+// {uuid} - required path parameter: ID of the identity to get
+func (s *service) GetIdentity(ctx context.Context, params *affiliation.GetIdentityParams) (i *models.IdentityDataOutput, err error) {
+	uuid := params.ID
+	log.Info(fmt.Sprintf("GetIdentity: uuid:%s", uuid))
```
Comment (lukaszgryglicki) [inline affiliation/service.go:1995]: Please change `uuid` to `id` here.

File: affiliation/service.go [hunk 2]
```diff
@@ -1983,6 +1987,27 @@ func (s *service) GetListProfiles(ctx context.Context, params *affiliation.GetLi
...
+	apiName, _, username, err := s.checkTokenAndPermission(params)
+	defer func() {
+		log.Info(fmt.Sprintf("GetIdentity(exit): apiName:%s username:%s err:%v", apiName, username, err))
+	}()
+	if err != nil{
+		return
+	}
+	i, err = s.shDB.GetIdentity(uuid, true, nil)
```
Comment (lukaszgryglicki) [inline affiliation/service.go:2003]: Please use `id` instead of `uuid` here.

File: affiliation/service.go [hunk 3]
```diff
@@ -1983,6 +1987,27 @@ func (s *service) GetListProfiles(ctx context.Context, params *affiliation.GetLi
...
+	if err != nil{
+		err = errs.Wrap(fmt.Errorf("Identity with UUID '%s' not found", uuid), apiName)
```
Comment (lukaszgryglicki) [inline affiliation/service.go:2005]: and here.

File: sh/curl_get_identity.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+export SKIP_PROJECT=1
+. ./sh/shared.sh
+if [ -z "$1" ]
+then
+  echo "$0: please specify profile UUID as a 2nd arg"
```
Comment (lukaszgryglicki) [inline sh/curl_get_identity.sh:6]: replace `profile UUID` with `identity ID`.

File: affiliation/service.go [hunk 5]
```diff
@@ -1983,6 +1987,27 @@ func (s *service) GetListProfiles(ctx context.Context, params *affiliation.GetLi
 	return
 }
 
+// GetIdentity: API params:
+// /v1/affiliation/get_identity/{uuid}
```
Comment (lukaszgryglicki) [inline affiliation/service.go:1991]: uuid -> id

File: affiliation/service.go [hunk 6]
```diff
@@ -1983,6 +1987,27 @@ func (s *service) GetListProfiles(ctx context.Context, params *affiliation.GetLi
...
+// {uuid} - required path parameter: ID of the identity to get
```
Comment (lukaszgryglicki) [inline affiliation/service.go:1992]: uuid -> id

File: sh/curl_get_identity.sh [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  exit 2
+fi
+uuid=$(rawurlencode "${1}")
```
Comment (lukaszgryglicki) [inline sh/curl_get_identity.sh:9]: uuid -> id

File: sh/curl_get_identity.sh [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+if [ ! -z "$DEBUG" ]
+then
+  echo curl -i -s -H "Origin: ${ORIGIN}" -H "Authorization: Bearer ${JWT_TOKEN}" -XGET "${API_URL}/v1/affiliation/get_identity/${uuid}"
```
Comment (lukaszgryglicki) [inline sh/curl_get_identity.sh:13]: 3 lines: uuid -> id

File: swagger/dev-analytics-affiliation.yaml [hunk 9]
```diff
@@ -615,6 +615,38 @@ paths:
         - $ref: '#/parameters/rows'
         - $ref: '#/parameters/page'
         - $ref: '#/parameters/q'
+  /affiliation/get_identity/{id}:
+    get:
+      summary: Get identity with given UUID
```
Comment (lukaszgryglicki) [inline swagger/dev-analytics-affiliation.yaml:620]: uuid -> id

**Review by lukaszgryglicki — CHANGES_REQUESTED** [review]
Please replace all possible uuid occurences with id.

File: affiliation/service.go (hunk 3 above)
Comment (linonymous) [reply affiliation/service.go]: done

File: sh/curl_get_identity.sh (hunk 4 above)
Comment (linonymous) [reply sh/curl_get_identity.sh]: done

File: affiliation/service.go (hunk 5 above)
Comment (linonymous) [reply affiliation/service.go]: done

File: sh/curl_get_identity.sh (hunk 7 above)
Comment (linonymous) [reply sh/curl_get_identity.sh]: done

File: sh/curl_get_identity.sh (hunk 8 above)
Comment (linonymous) [reply sh/curl_get_identity.sh]: done

File: swagger/dev-analytics-affiliation.yaml (hunk 9 above)
Comment (linonymous) [reply swagger/dev-analytics-affiliation.yaml]: done

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm

**Comment by lukaszgryglicki** [comment]
Merged & deployed, tested new API - works OK:
`API_URL=prod JWT_TOKEN=`cat secret/lgryglicki.token` ./sh/curl_get_identity.sh 0d5ce10b1e3bf5cf4c33da44b37a177a7833cdca | jq`:
```
{
  "email": "12628+lukewendling@users.noreply.github.com",
  "id": "0d5ce10b1e3bf5cf4c33da44b37a177a7833cdca",
  "last_modified": "2020-08-10T06:31:06.188Z",
  "name": "Luke Wendling",
  "source": "git",
  "uuid": "0d5ce10b1e3bf5cf4c33da44b37a177a7833cdca"
}
```

**Pull request merged**